src/core/lv_obj_pos.c

```diff
@@ -862,7 +863,9 @@ void lv_obj_invalidate(const lv_obj_t * obj)
     obj_coords.y2 += ext_size;
 
     lv_obj_invalidate_area(obj, &obj_coords);
-
+#if LV_USE_OBJ_DRAW_CACHE
+    lv_obj_draw_cache_invalidate(obj);
```

Comment:
It should be added rather to `lv_obj_invalidate_area()` as this the last invalidate-like function in the chain.